app-route.html

```diff
@@ -236,9 +237,8 @@
 
       __resetProperties: function() {
```

Comment:
@e111077 In previous discussions with @rictic we concluded that resetting `tail` would significantly impact performance of children. Since you reset the `tail` data, this means that all children will recalculate based on the data. In big applications this results in a waterfall of databinding updates, sometimes resulting in very long frames. This is the exact reason that right now `tail` is not altered when resetting a route.

I am okay for adding a property to enable/disable this behavior.